src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -1286,7 +1442,212 @@ static int _xccdf_policy_generate_fix_other(struct oscap_list *rules_to_fix, str
+				_write_text_to_fd(output_fd, command);
+				if (oscap_iterator_has_more(service_disable_it))
+					_write_text_to_fd(output_fd, ",");
+			}
+		}
+		if (oscap_iterator_has_more(service_enable_it)) {
+			_write_text_to_fd(output_fd, " --enabled=");
+			while (oscap_iterator_has_more(service_enable_it)) {
+				char *command = (char *) oscap_iterator_next(service_enable_it);
+				_write_text_to_fd(output_fd, command);
+				if (oscap_iterator_has_more(service_enable_it))
+					_write_text_to_fd(output_fd, ",");
+			}
+		}
+		_write_text_to_fd(output_fd, "\n\n");
+	}
+	oscap_iterator_free(service_disable_it);
+	oscap_iterator_free(service_enable_it);
+	return 0;
+}
+
+static int _generate_kickstart_packages(struct kickstart_commands *cmds, int output_fd)
+{
+	_write_text_to_fd(output_fd, "# Packages selection (required for security compliance)\n");
+	_write_text_to_fd(output_fd, "%packages\n");
+	/* openscap-scanner and scap-security-guide needs to be installed because we will run oscap in the %post section */
+	_write_text_to_fd(output_fd, "openscap-scanner\n");
+	_write_text_to_fd(output_fd, "scap-security-guide\n");
+	struct oscap_iterator *package_install_it = oscap_iterator_new(cmds->package_install);
+	while (oscap_iterator_has_more(package_install_it)) {
+		char *package = (char *) oscap_iterator_next(package_install_it);
+		_write_text_to_fd(output_fd, package);
+		_write_text_to_fd(output_fd, "\n");
+	}
+	oscap_iterator_free(package_install_it);
+	struct oscap_iterator *package_remove_it = oscap_iterator_new(cmds->package_remove);
+	while (oscap_iterator_has_more(package_remove_it)) {
+		char *package = (char *) oscap_iterator_next(package_remove_it);
+		_write_text_to_fd(output_fd, "-");
+		_write_text_to_fd(output_fd, package);
+		_write_text_to_fd(output_fd, "\n");
+	}
+	oscap_iterator_free(package_remove_it);
+	_write_text_to_fd(output_fd, "%end\n");
+	_write_text_to_fd(output_fd, "\n");
+	return 0;
+}
+
+static int _generate_kickstart_post(struct kickstart_commands *cmds, const char *profile_id, const char *input_path, int output_fd)
+{
+	_write_text_to_fd(output_fd, "%post\n");
+	char *dup = strdup(input_path);
+	char *basename = oscap_basename(dup);
+	free(dup);
+	char *oscap_command = oscap_sprintf(
+		"oscap xccdf eval --remediate --profile '%s' /usr/share/xml/scap/ssg/content/%s\n",
+		profile_id, basename);
+	free(basename);
+	_write_text_to_fd(output_fd, "# Perform OpenSCAP hardening (required for security compliance)\n");
+	_write_text_to_fd_and_free(output_fd, oscap_command);
+	struct oscap_iterator *post_it = oscap_iterator_new(cmds->post);
+	while (oscap_iterator_has_more(post_it)) {
+		char *command = (char *) oscap_iterator_next(post_it);
+		_write_text_to_fd(output_fd, command);
+		_write_text_to_fd(output_fd, "\n");
+	}
+	oscap_iterator_free(post_it);
+	_write_text_to_fd(output_fd, "%end\n");
+	_write_text_to_fd(output_fd, "\n");
+	return 0;
+}
+
+static char *_remove_slash(const char *in)
```

Comment:
Isn't `oscap_strrm` beter for that? https://github.com/OpenSCAP/openscap/blob/8f06d23de1845c4aaec9cd6b931405c1d5948101/src/common/util.h#L322

Reply:
Oh, okay, you actually need a copy. Well, I'd suggest a name like `_copy_path_without_slashes` then.